src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs

```diff
@@ -167,7 +156,7 @@ public CancellationToken(bool canceled) : this(canceled ? CancellationTokenSourc
         /// <exception cref="System.ArgumentNullException"><paramref name="callback"/> is null.</exception>
         public CancellationTokenRegistration Register(Action callback, bool useSynchronizationContext) =>
             Register(
-                s_actionToActionObjShunt,
+                new Action<object?>(static obj => ((Action)obj!)()),
```

Comment:
This won't actually allocate on every invocation, but I can't convince my brain of that, so I'd prefer if it were written without the `new` syntax, e.g.
```suggestion
                (Action<object?>)(static obj => ((Action)obj!)()),
```